pilot/cmd/pilot-agent/main.go

```diff
@@ -590,12 +590,22 @@ func waitForCerts(fname string, maxWait time.Duration) {
 }
 
 func main() {
+	ignoreUnknownFlagsAndWarn(proxyCmd)
```

Comment:
You should probably do this for the parent command as well, just for completeness.  Also, you can do this above when we declare the commands:

```go
= &cobra.Command{
		...
		FParseErrWhitelist: cobra.FParseErrWhitelist{
			// Allow unknown flags for backward-compatibility.
			UnknownFlags: true,
		},
		...
	}
```

Reply:
I considered both of those things, then problem is I wasn't sure of how to get a warning message with those two approaches.

If we do the parent command as well, then we get warnings for proxyCmd when we do other commands.
If we do the code you showed, we can't do that warnings I think.

If we don't care about the warnings then both of those are fine, I am just concerned someone will typo something and we will never know. 

Any ideas?

Reply:
Ok that makes sense